libres/lib/res_util/block_fs.cpp

```diff
@@ -562,42 +558,38 @@ block_fs_type *block_fs_mount(const fs::path &mount_file, int block_size,
     auto data_file = path / (base_name + ".data_0");
     auto index_file = path / (base_name + ".index");
     block_fs_type *block_fs;
-    {
 
-        if (!fs::exists(mount_file))
-            /* This is a brand new filesystem - create the mount map first. */
-            block_fs_fwrite_mount_info__(mount_file);
-        {
-            std::vector<long> fix_nodes;
-            block_fs = block_fs_alloc_empty(mount_file, block_size,
-                                            fsync_interval, read_only);
-
-            block_fs_open_data(block_fs, data_file);
-            if (block_fs->data_stream != nullptr) {
-                std::error_code ec;
-                fs::remove(index_file, ec /* error code is ignored */);
-                block_fs_build_index(block_fs, data_file, fix_nodes);
-            }
-            block_fs_fix_nodes(block_fs, fix_nodes);
+    if (!fs::exists(mount_file))
+        /* This is a brand new filesystem - create the mount map first. */
+        block_fs_fwrite_mount_info__(mount_file);
+    {
+        std::vector<long> fix_nodes;
+        block_fs = block_fs_alloc_empty(mount_file, block_size, fsync_interval,
+                                        read_only);
+
```

Comment:
The `fix_nodes` seems to always be empty so should move the `fix_nodes` into the function call:
```cpp
block_fs_fix_nodes(block_fs, std::vector<long>());
```

Reply:
To me it looks like the call to `block_fs_build_index(block_fs, data_file, fix_nodes);` fills it up before the call to `block_fs_fix_nodes`.
Am I reading this wrong?

Reply:
My bad, I grepped it wrong. Proceed :+1: